Ext docs tweeks (#1826)
kyma-project/busola: docs/extensibility/display-section.md

### Example

```json
[
  {
    "source": "spec.url",
    "sort": {
      "default": true,
      "compareFunction": "$compareStrings($first, $second)"
    }
  },
  { "source": "spec.priority", "widget": "Badge" },
  { "source": "$join(spec.toppings.name, ', ')" },
  {
    "name": "quantityIsMore",
    "source": "$filter(spec.toppings, function ($v, $i, $a) { $v.quantity > $average($a.quantity) })"
  },
  { "source": "$join(spec.volumes.name, ', ')" },
  {
    "source": "$filter(spec.volumes, function ($v, $i, $a) {'configMap' in $keys($v)})" // List the array of Volume objects that have a config map
  },
  {
    "source": "spec.volumes['configMap' in $keys($)]" // This is the alternative way of listing the array of Volume objects that have a config map
  },
  {
    "source": "$join(spec.volumes['configMap' in $keys($)].name, ', ')" // List volume names of volumes that have a config map
  }
]
```

## Resource _details_ overview

The **details** section defines the display structure for the details page.

> It contains three sections, `header`, `body`, and optional `resourceGraph`.

Reviewer: ```suggestion
It contains three components, `header`, `body`, and optional `resourceGraph`.
```

> The first two sections are a list of widgets to display in the **header** section and the body of the page respectively.

Reviewer: ```suggestion
The first two components are a list of widgets to display in the **header** section and the body of the page respectively.
```

> The `resourceGraph` section is used to configure the ResourceGraph which shows relationships between various resources.

Reviewer: ```suggestion
The `resourceGraph` component is used to configure the ResourceGraph which shows relationships between various resources.
```

### Header and body parameters

- **source** - contains a [JSONata](https://docs.jsonata.org/overview.html) expression used to fetch data for the widget. In its simplest form, it's the path to the value. Not required for presentational widgets.
- **name** - Name for the primary label of this field. Required for most widgets (except for some rare cases that don't display a label). This can be a key to use from the **translation** section.
- **widget** - optional widget to render the defined entry. By default the value is displayed verbatim.

> - **valuePreprocessor** - name of [value preprocessor](resources.md#value-preprocessors),

Reviewer: ```suggestion
- **valuePreprocessor** - name of [value preprocessor](resources.md#value-preprocessors).
```

{
  "details": {
    "resourceGraph": {
      "colorVariant": 2,
      "dataSources": [
        {
          "source": "relatedSecrets"
        },
        {
          "source": "relatedPizzaOrders"
        }
      ]
    }
  },
  "dataSources": {
    "relatedSecrets": {
      "resource": {
        "kind": "Secret",
        "version": "v1"
      },
      "filter": "$root.spec.recipeSecret = $item.metadata.name"
    },
    "relatedPizzaOrders": {
      "resource": {
        "kind": "PizzaOrder",
        "group": "busola.example.com",
        "version": "v1"
      },
      "filter": "$item.spec.pizzas[name = $root.metadata.name and namespace = $root.metadata.namespace]"
    }
  }
}
```

<img src="./assets/ResourceGraph.png" alt="Example of a ResourceGraph"  style="border: 1px solid #D2D5D9">

### Data scoping

Whenever an entry has both **source** and **children** properties, the **children** elements are provided with extra variables.

In the case of objects, a `$parent` variable contains the data of the parent element.

For example:

```json
[
  {
    "source": "spec",
    "widget": "Panel",
    "children": [{ "source": "$parent.entry1" }, { "source": "$parent.entry2" }]
  }
]
```

> will render the data for `spec.entry1` and `spec.entry2`.

Reviewer: ```suggestion
renders the data for `spec.entry1` and `spec.entry2`.
```
